src/main/scala/uk/gov/nationalarchives/omega/api/messages/reply/TypedIdentifier.scala

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2023 The National Archives
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package uk.gov.nationalarchives.omega.api.messages.reply
+
+import io.circe.{ Encoder, Json }
+import uk.gov.nationalarchives.omega.api.messages.reply.GenericIdentifierDerivation._
+import io.circe.syntax._
+
+case class TypedIdentifier(identifier: Identifier, identifierType: String)
```

Comment:
May I suggest a bit more Scaladoc describing the purpose of some of your case classes could be useful?

Reply:
Have added more Scaladoc and this led to some renaming as I am really trying to align the classes to the API schema and some of the class names didn't quite align.